src/H5Gent.c

```diff
@@ -130,11 +130,22 @@ H5G_ent_decode(const H5F_t *f, const uint8_t **pp, H5G_entry_t *ent)
     HDassert(pp);
     HDassert(ent);
 
+    if (H5_IS_BUFFER_OVERFLOW(*pp, ent->name_off, p_end))
+        HGOTO_ERROR(H5E_FILE, H5E_OVERFLOW, NULL, "image pointer is out of bounds")
```

Comment:
Ah I didn't catch this the first time but all the error values returned in this function should be updated to FAIL rather than NULL since H5G_ent_decode returns an herr_t